### PR TITLE
Update to ASP.NET Core 1.0 (Resolve #66)

### LINKED ISSUE #66: Migrate the DevExtreme data library Sample project to ASP.NET Core 1.0.0
Now the DevExtreme.AspNet.Data library has several incompatibility issues with the ASP.NET Core 1.0.0 release version.

For instance, the Microsoft.AspNetCore.Mvc.ModelBinding.ModelBindingResult.Success method signature was changed in 
[this commit](https://github.com/aspnet/Mvc/commit/88c9ae6588dd36b7a65d5066c0507dc3cec595a5#diff-2cbda9ff7d5a156fc478b38a437e518bL36) and the [DevExtreme.AspNet.Data.DataSourceLoadOptionsBinder.BindModelAsync](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/dd7e2941934f55aa14017ff3774f85adb145df88/net/Sample/DataSourceLoadOptions.cs#L20) method raises the System.MissingMethodException error.

### PR DESCRIPTION
**Pull request by San4es**

### CODE REVIEW

File: net/DevExtreme.AspNet.Data.Tests/project.json [hunk 1]
```diff
@@ -13,9 +13,9 @@
             "dependencies": {
                 "Microsoft.NETCore.App": {
                     "type": "platform",
-                    "version": "1.0.0-rc2-*"
+                    "version": "1.0.0"
                 },
-                "System.ComponentModel": "4.0.1-rc2-*",
+                "System.ComponentModel": "4.0.1",
                 "xunit": "2.1.0",
                 "dotnet-test-xunit": "1.0.0-rc2-*"
```
Comment (AlekseyMartynov) [inline net/DevExtreme.AspNet.Data.Tests/project.json]: use 2.2-preview2 version of donet-test-xunit

File: net/DevExtreme.AspNet.Data/project.json [hunk 2]
```diff
@@ -11,19 +11,23 @@
...
-        "Newtonsoft.Json": "5.0.1"
+        "Newtonsoft.Json": "5.0.1",
+        "NETStandard.Library": "1.6.0"
     },
 
     "frameworks": {
         "net40": { },
-        "netstandard1.0": {
-            "imports": [ "dnxcore50", "portable-net45" ],
+        "netstandard1.6": {
```
Comment (AlekseyMartynov) [inline net/DevExtreme.AspNet.Data/project.json]: This isn't right. We need to target lowest possible versoin, not highest.

File: net/Sample/Views/Home/GroupedList.cshtml [hunk 3]
```diff
@@ -10,14 +10,14 @@
                 store: DevExpress.data.AspNet.createStore({
                     loadUrl: "@Url.Action("Products", "Northwind")"
                 }),
-                group: "Category.CategoryName",
-                sort: "ProductName",
+                group: "category.categoryName",
```
Comment (AlekseyMartynov) [inline net/Sample/Views/Home/GroupedList.cshtml]: Any comments regarding the change of letter case?

**Comment by AlekseyMartynov** [comment]
Also: the 'install' section of appveyor.yml needs to be changed
```
  - appveyor DownloadFile https://raw.githubusercontent.com/dotnet/cli/v1.0.0-preview2/scripts/obtain/dotnet-install.ps1
  - powershell -File dotnet-install.ps1
```

**Comment by AlekseyMartynov** [comment]
Side note. For PRs, please use branches in your fork. Don't make PRs from master to master.

File: net/Sample/Views/Home/GroupedList.cshtml (hunk 3 above)
Comment (San4es) [inline net/Sample/Views/Home/GroupedList.cshtml]: @AlekseyMartynov 

Regarding [this](https://github.com/aspnet/Announcements/issues/194) breaking change, EF serializes JSON in this way. 

We have 2 options to fix this:
1) change letter case;
2) restore previous naming strategy using follow code:

     services
         .AddMvc()
         .AddJsonOptions(options => options.SerializerSettings.ContractResolver = new DefaultContractResolver());

I chose 1st option.

Comment (AlekseyMartynov) [inline net/Sample/Views/Home/GroupedList.cshtml]: :cool: 
Did you have a chance to check insert/update operations. Do the work after this change?

Comment (San4es) [inline net/Sample/Views/Home/GroupedList.cshtml]: Yes, I checked all demos and cases.

**Comment by San4es** [comment]
@AlekseyMartynov I fixed places which you pointed at.

**Comment by AlekseyMartynov** [comment]
I still cannot understand the purpose of changes in net/DevExtreme.AspNet.Data/project.json.
The project perfectly compiles and works without any of them.
Where is my mistake? :copyright:

**Comment by San4es** [comment]
@AlekseyMartynov it's very strange, because I have several "'Type' does not contain a definition for 'EmptyTypes'" errors with initial project.json version on my side.

I want to understand my mistake, too.

**Comment by AlekseyMartynov** [comment]
Let's debug together today.

**Comment by AlekseyMartynov** [comment]
:sheep: it

**Pull request merged**